src/main/java/seedu/address/logic/parser/ParserUtil.java

```diff
@@ -208,11 +224,10 @@ public static RecipeName parseRecipeName(String name) throws ParseException {
                     throw new ParseException(RecipeIngredientSet.MESSAGE_CONSTRAINTS);
                 }
                 Index index = parseIndex(ingredValues[0]);
-                IngredientQuantity qty = parseIngredientQuantity(ingredValues[1]);
+                IngredientQuantity qty = parseIngredientQuantityInRecipe(ingredValues[1]);
                 ingredientSet.put(index, qty);
             } catch (ParseException e) {
-                //Parsing index or ingredientQuantity throws parseException
-                throw new ParseException(RecipeIngredientSet.MESSAGE_CONSTRAINTS);
+                throw e;
```

Comment:
Why catch the exception to throw it back?

Reply:
parseIndex and parseIngredientQuantityInRecipe may also throw an exception. I was thinking of giving more specific error message, rather than standardising the RecipeIngredientSet.Constraints message to be thrown whenever something is entered wrongly